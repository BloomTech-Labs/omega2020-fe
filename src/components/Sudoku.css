.Sudoku {
	background-image: url('../images/Diamonds.jpg');
	height: 100vh;
	text-align: center;
	font-family: 'Roboto';
	font-size: 22px;
	height: 100%;
}

.Board {
	height: auto;
	width: 30vw;
	display: inline-block;
	margin: auto;
}

input {
	padding: 0;
	margin: 0;
	height: 40px;
	width: 40px;
	font-size: 28px;
	font-weight: bold;
	border-color: grey;
	background: #ffffff;
}

input[type='text']:disabled {
	background-color: #cdcac9;
}
input[type='text']:disabled {
	color: black;
}
input[type='text']:focus {
	background: #608cff;
}

.Square_Input {
	padding: 0;
	margin: 0;
	height: 100%;
	width: 100%;
}

.Square {
	border: 0.5px solid grey;
	padding: 0;
	width: 100%;
	height: 100%;
	text-align: center;
}

.Board_RowData {
	display: flex;
	width: 30vw;
	height: 7vh;
}

.SudokuButtons {
	display: inline-flex;
	margin-bottom: 2%;
}

.alert {
	padding: 20px;
	background-color: #608cff;
	color: black;
	margin-bottom: 15px;
	opacity: 1;
	transition: opacity 0.6s; /* 600ms to fade out */
}

.label svg {
	size: '30em';
}

.dark-mode .Sudoku {
	color: #fff;
	background-color: #313843;
	background-image: none;
	height: 100%;
}

.dark-mode input {
	color: #fff;
	background: slategrey;
	border-color: black;
}

.nature-mode .Sudoku {
	background-image: url('../images/forest.png');
	background-size: 100%;
	color: #fff;
	height: 100%;
}

.nature-mode input {
	color: #fff;
	background: slategrey;
	border-color: black;
}

.nature-mode input[type='text']:enabled {
	opacity: 0.8 !important;
	background-color: #acd052;
	color: black;
}

.nature-mode input[type='text']:disabled {
	opacity: 0.8 !important;
	background-color: #4c712f;
	color: white;
}

.nature-mode input[type='text']:focus {
	background: #f5f853;
}

.city-mode .Sudoku {
	color: #fff;
	background-image: url('../images/city.png');
	background-size: 100%;
	height: 100%;
}

.city-mode input {
	color: #fff;
	background: slategrey;
	border-color: black;
}

.city-mode input[type='text']:enabled {
	opacity: 0.8 !important;
	background-color: #cdd4d9;
	color: black;
}

.city-mode input[type='text']:disabled {
	opacity: 0.8 !important;
	background-color: black;
	color: white;
}

.city-mode input[type='text']:focus {
	background: #608cff;
}

.sky-mode .Sudoku {
	color: black;
	background-image: url('../images/sky.png');
	background-size: 100%;
	height: 100%;
}

.sky-mode input {
	color: #fff;
	background: slategrey;
	border-color: black;
}

.sky-mode input[type='text']:enabled {
	opacity: 0.8 !important;
	background-color: #b2ddf3;
	color: black;
}

.sky-mode input[type='text']:disabled {
	opacity: 0.8 !important;
	background-color: #163a4e;
	color: white;
}

.sky-mode input[type='text']:focus {
	background: #45bcf5;
}

@media only screen and (max-width: 600px) {
	.Board {
		width: 90%;
	}
	.Board_RowData {
		width: 100%;
	}
}
