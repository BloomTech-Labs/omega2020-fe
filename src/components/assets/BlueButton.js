import React from 'react';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { blue } from '@material-ui/core/colors';
import { withStyles } from '@material-ui/core/styles';

const BlueButton = (props) => {
  return (
    <ColorButton
      type={props.type}
      href={props.href}
      size='large'
      className='href'
      onClick={props.click}
    >
      <Typography variant='body2' className='title'>
        {props.title}
      </Typography>
    </ColorButton>
  );
};

const ColorButton = withStyles((theme) => ({
  root: {
    paddingLeft: 60,
    paddingRight: 60,
    paddingTop: 15,
    paddingBottom: 15,
    color: theme.palette.getContrastText(blue.A700),
    backgroundColor: blue.A700,
    '&:hover': {
      backgroundColor: blue[800],
    },
  },
}))(Button);

export default BlueButton;
