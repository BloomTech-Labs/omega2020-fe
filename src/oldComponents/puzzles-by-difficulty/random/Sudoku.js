import React, { useState, useEffect } from 'react';
import Board from '../../puzzle-builder/Board';
import { GetPuzzles } from './Puzzles';
import SudokuButtons from '../../SudokuButtons';
import axiosWithAuth from '../../../utils/axiosWithAuth';
import Settings from './../../themes/Settings';
import '../../Sudoku.css';

const Sudoku = () => {
  const [win, setWin] = useState(""); //Stores solution string here

  // Description of gameBoardState below
  // {
  //   boardState : "", => String of formated board values
  //   puzzleId: "", => Puzzle Id from DS and passed thru BE
  //   // time: 0, => Not yet implemented
  //   history   : [], => The history of current game play
  //   conflicts : new Set([])   => Array of conflicting values across fields (3x3 grid of cells), rows and columns.
  // });
  
  const [gameBoardState, setGameBoardState] = useState(
    {
      boardState : "",
      puzzleId: "",
      // time: 0,
      history   : [],
      conflicts : new Set([])  
    });
  
  
  // Retrieve puzzle data
  async function getRandomPuzzle() {
    var puzzles = await GetPuzzles();
    setWin(puzzles.solution);

    return puzzles;   // changed puzzle.sudoku to puzzles to return all the puzzles 
  };
  
  const getFormattedPuzzle = async () => {
    const puzzle = await getRandomPuzzle();
    const formattedPuzzle = formatPuzzle(puzzle.sudoku); // changed puzzles to puzzle.sudoku

    console.log("GBS in formatted puzzle", gameBoardState)
    console.log("Loaded puzzle in formatted puzzle", puzzle)
    console.log("formattedPuzzle  in formatted puzzle", formattedPuzzle);
      setGameBoardState({
        ...gameBoardState,
        puzzleId: puzzle.id,
        level: puzzle.level,
        boardState: formattedPuzzle,
        solved: puzzle.solution,
        original: puzzle.sudoku
      });
  };
  // Start the game here by getting a formatted puzzle
  useEffect(() => {
    getFormattedPuzzle();

  },[]) 

  function getDeepCopyOfArray(arr) {
    var now = JSON.parse(JSON.stringify(arr));
    console.log("NOW", now)
    return now;
  };

  const handleSquareValueChange = (i, j, newValue) => {
    setGameBoardState(prevState => {
      const newBoardState = getDeepCopyOfArray(prevState.boardState);
      const prevEditable = prevState.boardState[i][j].editable;
      newBoardState[i][j] = {
          cellValue : newValue,
          cellId    : stringify(i, j),
          editable  : prevEditable
        };

      // Now push the previous board state on the history stack
      const newHistory = getDeepCopyOfArray(prevState.history);
      newHistory.push(prevState.boardState);

      return {
        ...gameBoardState,
        boardState: newBoardState,
        history: newHistory, 
        conflicts: new Set([])
      };
    });
  };

  const handleUndoClick = () => {
    setGameBoardState(prevState => {
      const newHistory = getDeepCopyOfArray(prevState.history);
      const lastBoardState = newHistory.pop();

      // Now assign the previous board state as the current board state
      return {
        ...gameBoardState,
        boardState: lastBoardState, 
        history: newHistory,
        conflicts : new Set([])
      };
    });
  };
  
  const handleNewGameClick = () => {
    setGameBoardState({
      ...gameBoardState,
      boardState: getFormattedPuzzle(),
      history   : [],
      conflicts : new Set([])
    });
  };
  
  // ************** Saves sudoku state (data, difficulty, time) to backend *********


  const handleSaveClick = () => {
    console.log(gameBoardState);
    
    const puzzleId = gameBoardState.puzzleId;
    
    // Turn boardState into a string
    var playString = [];
    var playStringNow;
    
    for (var i=0; i<gameBoardState.boardState.length; i++) { // for each row
      for (var j=0; j<gameBoardState.boardState.length; j++) { // for each column
        playStringNow = gameBoardState.boardState[i][j].cellValue // the value in each cell
        playString.push(playStringNow)                // is pushed to playString
      };
      
    };
    // activePuzzleString = single string represents current board state
    // setActivePuzzleString(playString.join('')); 
    var activePuzzleString = playString.join(''); 

    const req = {
      // time: gameBoardState.time,
      difficulty: gameBoardState.level,
      data: activePuzzleString,
      solved: gameBoardState.solved,
      original: gameBoardState.original
      };
      console.log("WIN", win)
      
      axiosWithAuth()
      .post(`/user-puzzles/${puzzleId}`, req)
      .then(res => {

        console.log("AXIOS FROM SAVE CLICK", res);
        console.log("REQ", res);
        alert('Puzzle saved');
      });
    };
    // console.log("REQ2", req)
    console.log("WIN", win)
    console.log("GBS101", gameBoardState)


  function handleVerifyClick() {
    const { boardState, setBoardState } = gameBoardState;
    
    // Assigns id to boxes in two digit format for xy (row column)
    // rows[0]/cols[0] -> first row/column
    const rows = {};
    const cols = {};
    // Example: boxes['00'] -> an array of values in the first box. 
    const boxes = {};
    
    // populating rows
    for(let i=0; i<boardState.length; i++) {
      rows[i] = getDeepCopyOfArray(boardState[i]);
      // console.log("BOX ID: ", "boxId")
      
      for(let j=0; j<boardState[i].length;j++) {
        // populating columns
        if(cols.hasOwnProperty(j)) {
          cols[j].push(boardState[i][j]); //set a new cell value in the board
        } else {
          cols[j] = [boardState[i][j]]; // or keep the value
        };
        
        // populating boxes
        const boxId = stringify(Math.floor(i/3), Math.floor(j/3));
        
        if(boxes.hasOwnProperty(boxId)) {
          boxes[boxId].push(boardState[i][j]);
        } else {
          boxes[boxId] = [boardState[i][j]];
        };
      };
    };
    // creates an array of conflicts found by location
    const rowConflicts = flatten(getConflicts(Object.values(rows)));
    const colConflicts = flatten(getConflicts(Object.values(cols)));
    const boxConflicts = flatten(getConflicts(Object.values(boxes)));
    
    // console.log("BOX CONFLICTS1: ", boxConflicts)
    
    const mergedConflicts = [...rowConflicts, ...colConflicts, ...boxConflicts];
    setGameBoardState({...gameBoardState, conflicts: new Set(mergedConflicts)});
    
    // Turn boardState into a string
    var playString = [];
    var playStringNow;
    
    for (var i=0; i<gameBoardState.boardState.length; i++) { // for each row
      for (var j=0; j<gameBoardState.boardState.length; j++) { // for each column
        playStringNow = gameBoardState.boardState[i][j].cellValue // the value in each cell
        playString.push(playStringNow)                // is pushed to playString
      };
    };
    
    // activePuzzleString = single string represents current board state
    var activePuzzleString = playString.join(''); 
    console.log("activePuzzleString", activePuzzleString);
    console.log("WIN", win);
    // Check for Win when no conflicts are there
    
    if (mergedConflicts.length === 0){
      if (activePuzzleString === win){
        return (
          // build some animation for win here
          alert('Congratulations! You have solved the puzzle!')
          )};
        };
      };
      
      function flatten(a) {
        return Array.isArray(a) ? [].concat(...a.map(flatten)) : a;
      };
      
      function getConflicts(arrs) {
        return (arrs.map(arr => getConflictsInArray(arr)));
      };
      
      function getConflictsInArray(arr) {
        const conflictMap = {};
        
        for(let i=0; i<arr.length; i++) {
          let curr = arr[i];
          
          if(curr.cellValue !== ".") {
            if(conflictMap.hasOwnProperty(curr.cellValue)) {
              conflictMap[curr.cellValue].push(curr.cellId);
            } else {
              conflictMap[curr.cellValue] = [curr.cellId];
            };      
          };
        };
        
        return Object.values(conflictMap).filter(arr => arr.length>1); 
      };
      
      function formatPuzzle(puzzle) {
        const formattedPuzzle = createArray(9, 9);
        
        for(let i=0; i<puzzle.length; i++) {
          const rowId = getRowId(i);
          const colId = getColId(i);
          
          const editable = puzzle[i] === '.' ? true : false;
          
          formattedPuzzle[rowId][colId] = {
            cellValue : puzzle[i],
            cellId    : stringify(rowId, colId),
            editable  : editable
          };
        };
        return formattedPuzzle;
      };
      
      function stringify(num1, num2) {
        return num1 + '' + num2;
      };
      
      function getRowId(i) {
        return Math.floor(i/9);
      };
      
      function getColId(i) {
        return (i%9);
      };
      
      /*
      Returns a puzzle formatted like so:
      [
        [ob1, ob2, ob3],
        [.     .    . ],
        [.     .    . ],
      ]
      
      Where ob = {
        cellValue     : value of this cell,
        editable : true if this cell will be user defined, false otherwise
      }
      */  
     console.log("BOARD STATE BABYYYYY", gameBoardState)

     return (
       <div className = "Sudoku">
        <div>
          <SudokuButtons
            historyLength  = {gameBoardState.history.length}
            onUndoClick = {handleUndoClick}
            onNewGameClick = {handleNewGameClick}
            onVerifyClick  = {handleVerifyClick}
            onSaveClick = {handleSaveClick}
            />
        </div>
        
        <div>
          <Board
            className="Board"
            boardState = {gameBoardState.boardState}
            conflicts = {gameBoardState.conflicts}
            onSquareValueChange = {handleSquareValueChange}
            historyLength = {gameBoardState.history.length}
            />
        </div>  
        <Settings />
      </div>
    );
    
  };
  //
  function createArray(length) {
    var arr = new Array(length || 0),
        i = length;

    if (arguments.length > 1) {
        var args = Array.prototype.slice.call(arguments, 1);
        while(i--) arr[length-1 - i] = createArray.apply(this, args);
    };
    return arr;
};

export default Sudoku;