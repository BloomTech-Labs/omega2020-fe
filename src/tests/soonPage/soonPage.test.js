import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import SoonPage from '../../components/soonPage/soon';
import Adapter from 'enzyme-adapter-react-16';
import Sudoku from '../../components/soonPage/sudoku.png';
import BlueButton from '../../components/assets/BlueButton';

configure({ adapter: new Adapter() })

describe('all elements', () => {
    const soonPage = mount(
        <SoonPage />
    );

    it('snapshot the info of the soon page component', () => {
        const soonPageShot = renderer.create(<SoonPage />).toJSON();
        expect(soonPageShot).toMatchSnapshot();
      });

    it('renders firstFont element', () => {
        expect(
            soonPage.containsMatchingElement(
                <h1 className="firstFont"> WE KNOW YOU'RE EXCITED! </h1>
            )
        ).toBeTruthy()
    });

    it('renders the secondFont element', () => {
        expect(
            soonPage.containsMatchingElement(
                <p className="secondFont">Sudomega sudoku games are </p>
            )
        ).toBeTruthy()
    });

    it('renders the thirdFont element', () => {
        expect(
            soonPage.containsMatchingElement(
                <h1 className="thirdFont"> COMING SOON </h1>
            )
        ).toBeTruthy()
    })
})

describe('renders the image info such as alt and the src', () => {
    const soonPage = mount(
        <SoonPage />
    );

    const iSoonPage = soonPage.instance();

    it('renders all tags length', () => {
        const soonPage = shallow(
            <SoonPage />
        );
        expect(soonPage.find("div").length).toEqual(1)
        expect(soonPage.find("section").length).toEqual(1)
        expect(soonPage.find("h1").length).toEqual(2)
        expect(soonPage.find("p").length).toEqual(1)
    })


    it('renders the image', () => { 
        expect(soonPage.find("img").prop("src")).toEqual(Sudoku)
    })    

    it('renders the length of the image in the component', () => {
        expect(soonPage.find("img").length).toEqual(1)
    })
    
    it('renders the alt name', () => {
        const page = render(
            <SoonPage /> 
        );
        const alt = page.getByAltText(/sudoku/);
        expect(alt).toBeInTheDocument()
        expect(alt).toBeVisible()
        expect(alt).toBeTruthy()
    })

    it('shows button props', () => {
        const renderSoonPage = renderer.create(<SoonPage />); 
        const rootSoonPage = renderSoonPage.root;

        expect(rootSoonPage.findByType(BlueButton).props.title).toBe('learn how to play!')
        expect(rootSoonPage.findByType(BlueButton).props.href).toBe('/tutorial')


        expect(iSoonPage).toBe(null)
    })
})

it('render all className', () => {
    const { getByTestId } = render(
        <SoonPage />
    );
    expect(getByTestId('sudo-container')).toHaveClass('sudoku');
    expect(getByTestId('sudo-logo')).toHaveClass('sudoku-logo');
    expect(getByTestId('sudo-firstFont')).toHaveClass('firstFont');
    expect(getByTestId('sudo-secondFont')).toHaveClass('secondFont');
    expect(getByTestId('sudo-thirdFont')).toHaveClass('thirdFont');
    expect(getByTestId('sudo-button')).toHaveClass('button');
})