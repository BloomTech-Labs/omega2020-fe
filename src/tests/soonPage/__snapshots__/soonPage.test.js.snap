// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all elements snapshot the info of the soon page component 1`] = `
<div
  className="sudoku"
  data-testid="sudo-container"
>
  <br />
  <br />
  <img
    alt="sudoku"
    className="sudoku-logo"
    data-testid="sudo-logo"
    src="sudoku.png"
  />
  <h1
    className="firstFont"
    data-testid="sudo-firstFont"
  >
    WE KNOW YOU'RE EXCITED!
  </h1>
  <p
    className="secondFont"
    data-testid="sudo-secondFont"
  >
    Sudomega sudoku games are
  </p>
  <br />
  <br />
  <h1
    className="thirdFont"
    data-testid="sudo-thirdFont"
  >
    COMING SOON
  </h1>
  <br />
  <br />
  <section
    className="button"
    data-testid="sudo-button"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-text href MuiButton-textSizeLarge MuiButton-sizeLarge"
      href="/tutorial"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root title MuiTypography-body2"
        >
          learn how to play!
        </p>
      </span>
    </a>
  </section>
</div>
`;
